src/PointwiseFunctions/AnalyticSolutions/Elasticity/HalfSpaceMirror.hpp

```diff
@@ -0,0 +1,182 @@
+ * J_0(kr) k e^{(-kz)}\left(\frac{-2\mu}{\lambda + \mu}\right) \tilde{p}(k) \\
+ * S_{rr} &= \Theta - S_{\phi\phi} - S_{zz} \\
+ * S_{\phi\phi} &= \frac{\xi_{r}}{r} \\
+ * S_{(rz)} &= -\frac{1}{2 \mu} \int_0^{\infty} dk J_1(kr) k e^{(-kz)}\left(kz
+ * \right) \tilde{p}(k) \\
+ * S_{zz} &= \frac{1}{2 \mu} \int_0^{\infty} dk
+ * J_0(kr) k e^{(-kz)}\left(-\frac{\mu}{\lambda + \mu} - kz \right) \tilde{p}(k)
+ * \f}
+ *
+ * (Eqs. 11 and 12 in \cite Lovelace2007tn), where \f$ \tilde{p}(k)=
+ * \frac{1}{2\pi}e^{(\frac{kr}{2})^2}\f$ is the Hankel-Transform of the lasers
+ * intensity profile and \f$ \Theta = \mathrm{Tr}(S)\f$ the materials expansion.
+ *
+ */
+class HalfSpaceMirror {
+ public:
+  static constexpr size_t volume_dim = 3;
+
+  using constitutive_relation_type =
+      Elasticity::ConstitutiveRelations::IsotropicHomogeneous<3>;
+
+  struct BeamWidth {
+    using type = double;
+    static constexpr OptionString help{"The lasers beam width"};
+    static type lower_bound() noexcept { return 0.0; }
+  };
+
+  struct Material {
+    using type = constitutive_relation_type;
+    static constexpr OptionString help{"The material properties of the beam"};
+  };
+
+  struct IntegrationIntervals {
+    using type = size_t;
+    static constexpr OptionString help{
+        "Workspace size for numerical integrals"};
+    static type lower_bound() noexcept { return 1; }
+    static type default_value() noexcept { return 350; }
+  };
+
+  struct AbsoluteTolerance {
+    using type = double;
+    static constexpr OptionString help{
+        "Absolute tolerance for numerical integrals"};
+    static type lower_bound() noexcept { return 0.; }
+    static type default_value() noexcept { return 1e-12; }
+  };
+
+  struct RelativeTolerance {
+    using type = double;
+    static constexpr OptionString help{
+        "Relative tolerance for numerical integrals"};
+    static type lower_bound() noexcept { return 0.; }
+    static type upper_bound() noexcept { return 1.; }
+    static type default_value() noexcept { return 1e-10; }
+  };
+
+  using options = tmpl::list<BeamWidth, Material, IntegrationIntervals,
+                             AbsoluteTolerance, RelativeTolerance>;
+  static constexpr OptionString help{
+      "A semi-infinite mirror on which a laser introduces stress perpendicular "
+      "to the mirrors surface. The displacement then is the Hankel-Transform "
+      "of the general solution multiplied by the beam profile"};
+
+  HalfSpaceMirror() = default;
+  HalfSpaceMirror(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror& operator=(const HalfSpaceMirror&) noexcept = delete;
+  HalfSpaceMirror(HalfSpaceMirror&&) noexcept = default;
+  HalfSpaceMirror& operator=(HalfSpaceMirror&&) noexcept = default;
+  ~HalfSpaceMirror() noexcept = default;
+
+  HalfSpaceMirror(double beam_width,
+                  constitutive_relation_type constitutive_relation,
+                  size_t no_intervals = 350, double absolute_tolerance = 1e-12,
```

Comment:
better name for no_intervals? why this default?

Reply:
best make the variable name the same as the option name, i.e. `integration_intervals`

Reply:
The default values for the integration; i.e. no_intervals and tolerances I've chosen so that it would not fail for the domain size used in "Numerically modeling Brownian thermal noise in amorphous and crystalline thin coatings".